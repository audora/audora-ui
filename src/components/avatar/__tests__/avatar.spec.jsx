import React from 'react'
import { render, shallow } from 'enzyme'
import 'jest-styled-components'
import Avatar from '../avatar'
import Theme from '../../../theme'

const shallowWithTheme = tree => {
  const context = shallow(<Theme />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

describe('Avatar', () => {
  it('renders correctly', () => {
    const wrapper = shallowWithTheme(<Avatar name="João" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly when square', () => {
    const wrapper = render(
      <Theme>
        <Avatar name="João" square />
      </Theme>
    )
    // expect(wrapper).toHaveStyleRule('square')
    expect(wrapper.props().square).toBe(true)
  })

  it('renders image correctly', () => {
    const image = 'https://avatars3.githubusercontent.com/u/10627086?s=40&v=4'
    const wrapper = render(
      <Theme>
        <Avatar name="João" img={image} />
      </Theme>
    )
    expect(wrapper.props().img).toEqual(image)
  })

  it('renders all sizes correctly', () => {
    const testCases = [
      { size: 'tiny' },
      { size: 'small' },
      { size: 'medium' },
      { size: 'large' },
      { size: 'huge' },
    ]

    testCases.forEach(props => {
      const wrapper = render(
        <Theme>
          <Avatar {...props} />
        </Theme>
      )

      console.log('wrapper', wrapper)

      expect(wrapper.props().size).toEqual(props.size)
    })
  })
})
