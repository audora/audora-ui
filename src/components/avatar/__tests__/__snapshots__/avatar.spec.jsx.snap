// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<div
  class="sc-bwzfXH jYmbZe"
>
  J
</div>
`;
